commands/bench/bench.sh

```diff
@@ -121,12 +129,12 @@ main() {
   # but avoid committing them. It is so `cmd_runner_apply_patches` can work
   git restore --staged Cargo.*
 
-  git commit -m "$COMMIT_MESSAGE"
+  git commit -m "$COMMIT_MESSAGE" || :
 
   # Push the results to the target branch
   git remote add \
     github \
-    "https://token:${GITHUB_TOKEN}@github.com/${GH_CONTRIBUTOR}/${GH_CONTRIBUTOR_REPO}.git"
+    "https://token:${GITHUB_TOKEN}@github.com/${GH_CONTRIBUTOR}/${GH_CONTRIBUTOR_REPO}.git" || :
```

Comment:
These statements do not appear optional. Why do they return non-zero, and why do we want to continue execution if they do?

Reply:
it's a mess with these remotes. They supposed to be running only once in CI, but if I run them in loop as subshell they were not ending with zero but either 1 or 2, so it was stopping the process

Reply:
Okay, that explains `git remote add` / `git remote remove`, but why do we want to continue if `git commit` fails?

Reply:
because "Nothin to commit" results `1` too, and stops the subshell thus loop above

Reply:
If we want to do nothing in case where there's no changes to commit and push, maybe explicitly return early? 
Something like
```bash
  # Commits the weights and pushes it
  git add .
  # instead of using `cargo run --locked`, we allow the Cargo files to be updated
  # but avoid committing them. It is so `cmd_runner_apply_patches` can work
  git restore --staged Cargo.*

  if [ -z "$(git --no-pager diff --name-only --cached)" ]; then
    echo "Nothing to commit; weights did not change"
    return
  fi

  git commit -m "$COMMIT_MESSAGE"
```

Reply:
I think the way it's done should also be okay. You think no?

Reply:
It's okay from the execution standpoint, I just couldn't understand the intentions.  
I definitely couldn't guess that it's "noting to commit" case, just from the code alone.

Reply:
ahh, got it. I will add explanation there 👍